### ISSUE TITLE
[Chrony] - ServerStats: handle ServerStats4 Response

### ISSUE THREAD

**Issue by Frankkkkk** [influxdata/telegraf#15553]
### Relevant telegraf.conf

```toml
[[inputs.chrony]]
    metrics = ["serverstats"]
```

### Logs from Telegraf

```text
2024-06-21T22:49:15Z E! [inputs.chrony] Error in plugin: querying server statistics failed: not implemented reply type 25 from &{Version:6 PKTType:reply Res1:0 Res2:0 Command:54 Reply:25 Status:SUCCESS Pad1:0 Pad2:0 Pad3:0 Sequence:1 Pad4:0 Pad5:0}
```

### System info

1.32 - 78cbf533d2

### Steps to reproduce

1. Use the chrony plugin + serverstats metric
2. see it fail on new chrony versions >= 4.4

### Expected behavior

It should work

### Actual behavior

It fails

### Additional info

The serverstats response was changed in chrony 4.4.

Waiting on https://github.com/facebook/time/pull/367

**Comment by powersj** [comment]
@Frankkkkk for this one, we are on hold till the upstream PR you linked to is landed?

**Comment by Frankkkkk** [comment]
@powersj yes, IMHO that's best. I'll create a PR once the upstream's PR is merged :)

**Comment by crmdias** [comment]
Hi all! Thanks for the report. This has been addressed with commit 18207c. Please verify so we can close this issue.

**Comment by Frankkkkk** [comment]
Thanks for the heads up @crmdias . I just saw the small "typo" with capital `Timestamp` vs smallcaps `timestamp`.. Well, we can fix that in the future :-)

**Comment by crmdias** [comment]
Hi! I have replied to the comment on the original PR. Please let me know if there are still any concerns or follow ups needed.

> Thanks for pointing this out @Frankkkkk ! Could you please review the [commit that was pushed](https://github.com/facebook/time/commit/18207c5d8ddc7242e8d4192985898b6dbe66932c)? I believe the change addressed this. This PR was closed without merging, as my commit was generated by internal automation with no accompanying PR.
